### ISSUE TITLE
WebIO on JuliaBox

### ISSUE THREAD

**Issue by rdeits** [JuliaGizmos/WebIO.jl#88]
@shashi I know we chatted about this on Slack, but I just wanted to create an issue to track the status. So far, I've been trying to run WebIO on JuliaBox (using the new https://next.juliabox.com/ ), but without any success. 

So far I've tried the following:

* Added WebIO through the JuliaBox package manager
* Manually invoked WebIO's `build.jl` script to create `~/.jupyter/jupyter_notebook_config.py` 
* Also copied the config file to `/juliabox/.jupyter/` which the juliabox logs suggest might be the folder it's actually using
* Logged out and back in to restart the server a few times

but no luck. I'm still getting `GET https://next.juliabox.com/pkg/WebIO/webio/dist/bundle.js?_=1522301626124 404 (Not Found)` in the JS console. I've also tried poking around at other paths, as in #75 but I can't find the webio JS being served anywhere. 

Just to make sure the only problem was serving the files, I uploaded WebIO's `bundle.js` and `ijulia_setup.js` to juliabox and then rendered them with:

```julia
display(MIME("text/javascript"), open(readstring, "bundle.js"))
display(MIME("text/javascript"), open(readstring, "ijulia_setup.js"))
```

that worked well enough to get basic WebIO nodes rendering, although the imports in my WebIO Scope still don't load, since they're looking for the same `/pkg/` URLs.

**Comment by shashi** [comment]
Could also be #58 ...

cc @mdpradeep

**Comment by rdeits** [comment]
Good point. I just checked the juliabox server, and I don't see a `jupyter_notebook_config.json`. However, it does look like the `jupyter_notebook_config.py` that I manually created in `/juliabox/.jupyter/` seems to have disappeared. Perhaps that folder gets re-initialized on boot?

**Comment by rdeits** [comment]
JuliaBox is still using Jupyter in notebook (not jupyterlab mode), so I think the jupyter labextension shouldn't be needed. Unfortunately, `Pkg.build("WebIO")` isn't possible because JuliaBox disables all `Pkg` operations. I was able to run the WebIO build by manually invoking the build.jl file, but it still doesn't seem to have installed the extension.

**Comment by djsegal** [comment]
I would still try to do the labextension trick. 

I don't think I use labs for anything and it's resolved the issue you're having for simple notebooks (locally and in production)

**Comment by shashi** [comment]
@mdpradeep @tanmaykm would be great if you can investigate this.

**Comment by mdpradeep** [comment]
Was checking this @shashi, There is quite a bit of dependence on the `Pkg` operations which are not supported on JuliaBox. For e.g. https://github.com/JuliaGizmos/WebIO.jl/blob/master/deps/_devsetup.jl and https://github.com/JuliaGizmos/WebIO.jl/blob/master/deps/_bundlejs.jl do many `Pkg` operations. We need to find solutions to address them.

**Comment by rdeits** [comment]
@djsegal I did try your suggestion, but juliabox doesn't have jupyterlab, so `jupyter labextension install webio` gives `Error executing Jupyter command 'labextension': [Errno 2] No such file or directory`. 

@mdpradeep assuming the Pkg issues can be resolved, are jupyter server extensions with `jupyter_notebook_config.py` supported on juliabox?

**Comment by shashi** [comment]
@mdpradeep we don't need to run those scripts to build. Those are only for someone _developing_ WebIO.

See build.jl which doesn't have any Pkg calls.

**Comment by xorJane** [comment]
Yes, thanks @mpradeep!
